Translating/ModelSBML.py

```diff
@@ -0,0 +1,315 @@
+        new_species.setConstant(False)
+        new_species.setInitialAmount(0)
+        self.create_species_features(comp_agent, new_species_multi_plugin)
+
+        new_species.setId(comp_agent.to_SBML_species_code())
+        new_species.setName(str(comp_agent))
+
+    def create_all_species_compartments_and_complex_species_types(self, unique_complexes: dict):
+        """
+        Creates all compartments
+        Creates species for every unique Complex agent in values
+        Creates SpeciesType for Complex agent if there is new one - no isomorphisms here
+
+        :param: unique_complexes -> dict() of representative Complexes[Keys]
+                                    & tuple() of pairs (Core.Complex,str. SBML_species_code())
+                                    Encapsulates all used Complexes and theirs isomorphisms in the model
+        """
+        for comp_agent in unique_complexes:
+            # Not composed agents are created 2 times
+            # firstly as structs from signature and secondly as complex here
+            # create species type from complex agent if it does not exist.
+            if comp_agent.to_SBML_speciesTypes_code() not in self.finishedComplexTypes:
+                self.create_species_type_from_complex(comp_agent)
+                self.finishedComplexTypes.append(comp_agent.to_SBML_speciesTypes_code())
+
+            for agent in unique_complexes[comp_agent]:
+                if agent[0].compartment not in self.finishedCompartments:
+                    self.create_compartment(agent[0].compartment)
+                self.create_species(agent[0])
+
+    def create_reactants(self, items:list(), reaction):
+        """Creates all reactants from Complexes in list of tuples
+        :param: items -> list of LHS complexes in reaction
+        """
+        for itm in items:
+            reactant = reaction.createReactant()
+            reactant.setSpecies(itm[0].to_SBML_species_code())
+            reactant.setConstant(False)
+            reactant.setStoichiometry(itm[1])
+
+    def create_products(self, items:list(), reaction):
+        """Creates all products from Complexes in list of tuples
+        :param: items -> list of RHS complexes in reaction"""
+        for itm in items:
+            product = reaction.createProduct()
+            product.setSpecies(itm[0].to_SBML_species_code())
+            product.setConstant(False)
+            product.setStoichiometry(itm[1])
+
+    def create_kinetic_law_and_modifiers(self, rules, reaction_objects):
+        """creates kinetic law for all reactions
+           and at the same time it creates modifiers
+           for those Complexes that were not used either
+           as reactant or product.
+          :param: rules -> BCSL rules
+          :param: reaction_objects -> list of all libsbml reaction objects inorder
+           to work with given reaction in this librialry
+
+         """
+        for i, r in enumerate(rules):
```

Comment:
Good point. I think that in this case it does not create a problem because i do not rely on ordering, enumerate is just missused for asigning specific ID for every reaction. But maybe it could be done nicer

Reply:
You are actually relying on the ordering by accessing individual reactions in `reaction_objects`.

Reply:
True, resolved